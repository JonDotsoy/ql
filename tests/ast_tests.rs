#[cfg(test)]
mod ast_tests {
    use insta::assert_debug_snapshot;
    use ql::{ast::AST, Tokenizer};

    #[test]
    fn ast_parse_identifier_1() {
        let payload = "$.a";
        let tokens = Tokenizer::lexer(payload).unwrap();
        assert_debug_snapshot!(AST::parse(tokens));
    }

    #[test]
    fn ast_parse_identifier_2() {
        let payload = "$.a.b";
        let tokens = Tokenizer::lexer(payload).unwrap();
        assert_debug_snapshot!(AST::parse(tokens));
    }

    #[test]
    fn ast_parse_identifier_3() {
        let payload = r#"$.a["as\"d"]"#;
        let tokens = Tokenizer::lexer(payload).unwrap();
        assert_debug_snapshot!(AST::parse(tokens));
    }
}
