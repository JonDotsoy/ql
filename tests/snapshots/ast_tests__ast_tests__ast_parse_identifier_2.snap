---
source: tests/ast_tests.rs
assertion_line: 17
expression: "AST::parse(tokens)"
---
AST(
    Expression {
        span: Span {
            start: 0,
            end: 5,
        },
        expresssion: [
            MemberExpression {
                span: Span {
                    start: 0,
                    end: 5,
                },
                object: Identifier {
                    span: Span {
                        start: 0,
                        end: 1,
                    },
                    value: String(
                        "$",
                    ),
                },
                property: MemberExpression {
                    span: Span {
                        start: 2,
                        end: 5,
                    },
                    object: Identifier {
                        span: Span {
                            start: 2,
                            end: 3,
                        },
                        value: String(
                            "a",
                        ),
                    },
                    property: Identifier {
                        span: Span {
                            start: 4,
                            end: 5,
                        },
                        value: String(
                            "b",
                        ),
                    },
                },
            },
        ],
    },
)
